From: Antonio Terceiro <terceiro@debian.org>
Date: Sat, 7 May 2022 13:50:25 -0300
Subject: test/helper.rb: ignore GC compaction on unsupported platforms

For example, ruby 3.0 on Debian ppc64el architecture does not support GC
compaction. When running the tests, every 20th test crashes like this:

> Error:
> Minitest::Result#test_parsing_attribute_namespace:
> NotImplementedError: Compaction isn't available on this platform
>     <internal:gc>:213:in `compact'
>     /home/terceiro/ruby-nokogiri-1.13.5+dfsg/test/helper.rb:123:in `teardown'
---
 test/helper.rb | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/test/helper.rb b/test/helper.rb
index 8d4569e..4351aea 100644
--- a/test/helper.rb
+++ b/test/helper.rb
@@ -191,14 +191,20 @@ module Nokogiri
           GC.start(full_mark: true)
         when "compact"
           if @@test_count % COMPACT_EVERY == 0
-            GC.compact
+            begin
+              GC.compact
+            rescue NotImplementedError
+            end
           else
             GC.start(full_mark: true)
           end
         when "verify"
           if @@test_count % COMPACT_EVERY == 0
             # https://alanwu.space/post/check-compaction/
-            gc_verify_compaction_references
+            begin
+              gc_verify_compaction_references
+            rescue NotImplementedError
+            end
           end
           GC.start(full_mark: true)
         when "stress"
